XenAdmin/Commands/RemoveHostFromPoolCommand.cs

```diff
@@ -121,38 +120,40 @@ protected override void ExecuteCore(SelectedItemCollection selection)
             RunMultipleActions(actions, Messages.REMOVING_SERVERS_FROM_POOL, Messages.POOLCREATE_REMOVING, Messages.POOLCREATE_REMOVED, true);
         }
 
+        protected override string GetCantExecuteReasonCore(IXenObject item)
+        {
+            if (item is Host host)
+            {
+                if (host.IsMaster())
+                    return Messages.MESSAGEBOX_POOL_MASTER_REMOVE;
+
+                if (!host.IsLive())
+                    return Messages.HOST_UNREACHABLE;
+
+                if (host.resident_VMs != null && host.resident_VMs.Count > 1)

```

Comment:
Could consider `host.resident_VMs.Any()` instead of `host.resident_VMs.Count > 1`.

Reply:
`Any()` means `>0`, but it has to be `>1` because the first resident VM is dom0 which need to be excluded from the check.

Reply:
Good point, I missed that.